modules/services/heartbeat_monitor.lua

```diff
@@ -0,0 +1,134 @@
+local events = require('events')
+local constants = require('protocol_handler/ford_protocol_constants')
+ 
+ local module = {}
+ local mt = { __index = { } }
+  
+ 
+function mt.__index:PreconditionForStartHeartbeat()
+    local event = events.Event()
+    event.matches = function(s, data)
+      return data.frameType == constants.FRAME_TYPE.CONTROL_FRAME and
+      data.serviceType == constants.SERVICE_TYPE.CONTROL and
+      data.frameInfo == constants.FRAME_INFO.HEARTBEAT   and
+      self.session.sessionId == data.sessionId
+    end
+    self.expectations:ExpectEvent(event, "Heartbeat")
+    :Pin()
+    :Times(AnyNumber())
+    :Do(function(data)
+        if self.heartbeatEnabled and self.answerHeartbeatFromSDL then
+          self.control_services:Send( { serviceType = constants.SERVICE_TYPE.CONTROL,
+              frameInfo = constants.FRAME_INFO.HEARTBEAT_ACK } )
+        end
+      end)
+
+    local d = qt.dynamic()
+    self.heartbeatToSDLTimer = timers.Timer()
+    self.heartbeatFromSDLTimer = timers.Timer()
+
+    function d.SendHeartbeat()
+      if self.heartbeatEnabled and self.sendHeartbeatToSDL then
+        self.control_services:Send( {
+            serviceType = constants.SERVICE_TYPE.CONTROL,
```

Comment:
@LuxoftAKutsan serviceType in control_services:Send can be not just CONTROL, it can be also PCM, VIDEO and BULK_DATA or anything, except RPC. Now serviceType cannot be setup explicitly in control_services.